### PR TITLE
gui: add Input and Spacer widgets

### PR DESCRIPTION
**Pull request by casaca24**

### CODE REVIEW

**Comment by usaoc** [comment]
Thank you for the PR!  Ideally, the change will need documentation (even preliminary, gross ones will be fine), under `rhombus/gui/scribbling`.

File: rhombus/gui/private/input.rhm [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      Size
+      Stretch
+  "view.rhm".View
+
+export:
+  Input
+
+
+
+
```
Comment (usaoc) [inline rhombus/gui/private/input.rhm:28]: Can you trim the blank lines here?

File: rhombus/gui/private/spacer.rhm [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#lang rhombus/static/and_meta
+import:
+  lib("racket/gui/easy.rkt")
+  "view.rhm".View
+
+
+
+
```
Comment (usaoc) [inline rhombus/gui/private/spacer.rhm:8]: Ditto.

File: rhombus/gui/private/input.rhm [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+class Input(hand, private _at_content :: Obs):
+  implements View
+  internal _Input
+  opaque
+  override property handle: hand
+
+  constructor (content :: MaybeObs.of(String) = "",
```
Comment (usaoc) [inline rhombus/gui/private/input.rhm:35]: I think this annotation isn’t quite right—see the note on `#:value->text`.

File: rhombus/gui/private/input.rhm [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+               ~action: action :: maybe(Function.of_arity(2)) = #false,
+	       ~label: label :: MaybeObs.of(LabelString) = "",
+               ~is_enabled: is_enabled :: MaybeObs.of(Boolean) = #true,
+	       ~background_color: bg_color :: MaybeObs.of(maybe(Color)) = #false,
+	       ~style: style :: MaybeObs.of(List.of(Input.StyleSymbol)) = [#'single],
+	       ~font : font :: Font = normal_control_font,
+	       // ~keymap: ... TODO
+	       ~margin: margin :: MaybeObs.of(Margin) = [0, 0],
+               ~min_size: min_size :: MaybeObs.of(Size) = [#false, #false],
+               ~stretch: stretch :: MaybeObs.of(Stretch) = [#false, #false],
```
Comment (usaoc) [inline rhombus/gui/private/input.rhm:45]: The default value is wrong; should be `[#true, #true]` as in `gui-easy`.

**Comment by casaca24** [comment]
> Thank you for the PR!  Ideally, the change will need documentation (even preliminary, gross ones will be fine), under `rhombus/gui/scribblings`.

Thanks for the input!
I'll work on all your suggestions, and start working on the docs. This PR could wait until documentation is ready

File: rhombus/gui/private/input.rhm (hunk 3 above)
Comment (casaca24) [reply rhombus/gui/private/input.rhm:35]: Should the annotation be ommited? Like in `canvas.shm`?
```suggestion
  constructor (content,
```

Comment (usaoc) [reply rhombus/gui/private/input.rhm:35]: Omitting it is fine (conceptually, it’s `MaybeObs.of(Any)`, which is just `Any` anyway, but the docs can mention the former instead).

**Comment by usaoc** [comment]
Sorry, why was this closed?  If you just want to update the PR, you can force push to the will-be-merged branch.

**Comment by casaca24** [comment]
My bad, I'll try it next time.

**Comment by casaca24** [comment]
> Sorry, why was this closed? If you just want to update the PR, you can force push to the will-be-merged branch.

I had already deleted the previous fork when I read your message. Thanks for your help, I submitted another pull request.

**Pull request closed without merging**